bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/RuntimeUtils.java

```diff
@@ -107,6 +108,56 @@ public static boolean isByteLiteral(int intValue) {
         return (intValue >= BBYTE_MIN_VALUE && intValue <= BBYTE_MAX_VALUE);
     }
 
+    /**
+     * Decode the encoded Type name for runtime name calls.
+     *
+     * @param typeName Name of the defined type
+     * @return decoded name String
+     */
+    public static String decodeTypeName(String typeName) {
+        if (typeName == null) {
+            return typeName;
+        }
+        StringBuilder sb = new StringBuilder();
+        int index = 0;
+        while (index < typeName.length()) {
+            if (typeName.charAt(index) == '$' && index + 4 < typeName.length()) {
```

Comment:
we can bring this if into while loop condition

Reply:
This is not possible because we have an `else` condition inside the loop.